src/status_im2/contexts/syncing/enter_sync_code/view.cljs

```diff
@@ -0,0 +1,79 @@
+(ns status-im2.contexts.syncing.enter-sync-code.view
+  (:require [clojure.string :as string]
+            [quo2.core :as quo]
+            [quo2.foundations.colors :as colors]
+            [reagent.core :as reagent]
+            [react-native.core :as rn]
+            [react-native.clipboard :as clipboard]
+            [utils.i18n :as i18n]
+            [status-im2.contexts.syncing.enter-sync-code.style :as style]
+            [utils.debounce :as debounce]
+            [utils.re-frame :as rf]
+            [status-im2.contexts.syncing.utils :as sync-utils]))
+
+(defn view
+  []
+  (let [default-value           (reagent/atom nil)
+        submit-button-disabled? (reagent/atom true)
+        profile-color           (rf/sub [:profile/customization-color])]
+    (fn []
+      (let [invalid?           false
+            show-paste-button? (string/blank? @default-value)]
+        [:<>
+         [rn/view
+          {:style style/label-container}
+          [quo/text
+           {:style  style/label-pairing
+            :weight :medium
+            :size   :paragraph-2}
+           (i18n/label :t/type-pairing-code)]]
+         [rn/view {:style style/container-text-input}
+          [rn/view {:style (style/text-input-container invalid?)}
+           [rn/text-input
+            {:style                  (style/text-input)
+             :default-value          @default-value
+             :placeholder            (i18n/label :t/type-some-chat-key)
+             :on-change-text         (fn [scan-code]
+                                       (reset! default-value scan-code)
+                                       (reset! submit-button-disabled? false))
+             :blur-on-submit         true
+             :return-key-type        :done
+             :accessibility-label    :enter-sync-code-input
+             :auto-capitalize        :none
+             :placeholder-text-color colors/white-opa-40
+             :multiline              true}]
+           (if show-paste-button?
+             [rn/view {:style style/button-paste}
+              [quo/button
+               {:on-press (fn [_]
+                            (clipboard/get-string #(reset! default-value %))
```

Comment:
does `default-value` really works? because it's set only once on init, on render it should be ignore by component

Reply:
just wondering

Reply:
the `default-value` atom is used to keep track of input value and used here  
https://github.com/status-im/status-mobile/blob/add-scan-sync-code-tab/src/status_im2/contexts/syncing/enter_sync_code/view.cljs#L70-L72

is the name of the atom misleading or something?

Reply:
but yes reseting `default-value` also updates the text-input, I tested it on simulator :)

Reply:
the point of the default value is to be set only once, so, something is wrong here

Reply:
seems like we can change it @flexsurfer, ref: https://reactnative.dev/docs/0.69/textinput#defaultvalue
I also see this pattern being used in other parts of the codebase.
ref : https://github.com/status-im/status-mobile/blob/develop/src/quo2/components/wallet/token_input/view.cljs#L43-L44

Reply:
there is `value` prop for that, but we should avoid using it, because it's laggy. default value won't have any effect on re-render, it's used only once on component init, we could use `value` here, because i don't think anyone would try to type here , usually it will be paste

Reply:
`wallet.token-input` implementation looks strange

Reply:
so a recap on what is going on : 
- on init we set `default-value` of text-input component to a reagent atom ( we call it `default-value`) that is `nil`
- when user taps in this text-input `on-change` event is fired which updates this atom.
- when or if user presses paste, we then update this atom which also updates `default-value` of the text-component
- when we press clear icon, we clear this atom and also the `default-value`
- we do not rely on `value` prop of text-input component because it causes lag

So all in all this looks fine to me :)

Reply:
its a reagent thingy : https://cljdoc.org/d/reagent/reagent/1.2.0/doc/tutorials/controlled-inputs#react-native

Reply:
![Screenshot 2023-08-07 at 5 24 46 PM](https://github.com/status-im/status-mobile/assets/64726664/a783480b-5b34-4c61-91e3-348c5620b30a)

Reply:
@flexsurfer : please read https://github.com/reagent-project/reagent/issues/485#issuecomment-613850939

Reply:
yes you confirmed my statement, i don't see you updating the key for the force reinitialization, i mean , let's just use value, for this component it will work fine , using default-value with force reinitialization is hack which we don't need here, but i still don't know why it works now even without key hack, it doesn't make sense, because the point of this field is to set it only once

Reply:
yes we should move on but with the proper solution, currently, we have a few places in the code with the wrong usage of `default-value` so we should fix it , but not continue to use the wrong solution, again in you link key is updated for force refresh, we don't have it in our code

Reply:
found a solution that works well with `value` see : https://github.com/status-im/status-mobile/pull/16852/commits/7729f0a2d8ec3e258acf2490559d0f7bb05b39cb 
Will create issue to get rid of `default-value` hack as well.

Reply:
https://github.com/status-im/status-mobile/issues/16909

Reply:
@smohamedjavid : https://github.com/status-im/status-mobile/issues/16910 issue for making sure add contacts UX is similar